web/src/classic/components/organisms/Authentication/RootPage/hooks.ts

```diff
@@ -19,54 +14,20 @@ export default () => {
   const navigate = useNavigate();
   const { isAuthenticated, isLoading, error: authError, login, logout } = useAuth();
   const [error, isErrorChecked] = useCleanUrl();
-  const [currentWorkspace, setWorkspace] = useSessionWorkspace();
+  const [currentWorkspace, setCurrentWorkspace] = useWorkspace();
   const [currentUserId, setCurrentUserId] = useUserId();
   const [, setNotification] = useNotification();
-  const [lastWorkspace, setLastWorkspace] = useWorkspace();
 
   const { data, loading } = useGetTeamsQuery({ skip: !isAuthenticated });
 
   if (isAuthenticated && !currentUserId) {
     setCurrentUserId(data?.me?.id);
   }
-  useEffect(() => {
-    if (!currentWorkspace && lastWorkspace) setWorkspace(lastWorkspace);
-  }, [currentWorkspace, lastWorkspace, setWorkspace]);
 
   const workspaceId = useMemo(() => {
     return currentWorkspace?.id || data?.me?.myTeam.id;
   }, [currentWorkspace?.id, data?.me?.myTeam.id]);
 
-  const handleRedirect = useCallback(() => {
-    if (currentUserId === data?.me?.id) {
-      setWorkspace(
-        workspaceId
-          ? data?.me?.teams.find(t => t.id === workspaceId) ?? data?.me?.myTeam
-          : undefined,
-      );
-      setLastWorkspace(currentWorkspace);
-
-      navigate(`/dashboard/${workspaceId}`);
-    } else {
-      setCurrentUserId(data?.me?.id);
```

Comment:
I think this function had two functionalities.

a. Handle redirect
b. Set user id if `currentUserId !== data?.me?.id`

This modification looks like losing `b`, is it merely unnecessary code?

It might be necessary when switching accounts.